File: domain/src/main/java/com/arturogutierrez/openticator/domain/account/repository/AccountDataStore.java
package com.arturogutierrez.openticator.domain.account.repository;

import com.arturogutierrez.openticator.domain.account.model.Account;
import com.arturogutierrez.openticator.domain.category.model.Category;
import java.util.List;
import rx.Observable;

public interface AccountDataStore {

  Observable<Account> add(Account account);

  Observable<Account> update(Account account);

  Observable<Void> remove(Account account);

  Observable<List<Account>> getAccounts(Category category);

  Observable<List<Account>> getAllAccounts();
}

File: domain/src/main/kotlin/com/arturogutierrez/openticator/domain/account/repository/AccountDataStore.kt
package com.arturogutierrez.openticator.domain.account.repository

import com.arturogutierrez.openticator.domain.account.model.Account
import com.arturogutierrez.openticator.domain.category.model.Category
import io.reactivex.Completable
import io.reactivex.Flowable
import io.reactivex.Single

interface AccountDataStore {

  fun add(account: Account): Single<Account>

  fun update(account: Account): Single<Account>

  fun remove(account: Account): Completable

  fun getAccounts(category: Category): Flowable<List<Account>>

  val allAccounts: Flowable<List<Account>>
}
